Support installing the previous stable release

abhinav opened the pull request:
This adds support for installing the previous stable release of Go with
`gimme oldstable`.

Coupled with `gimme stable`, this will make it possible for projects to easily
support the current and previous stable releases of Go. This matches
[Go's Release Policy][1]: the current and previous stable releases are
supported until a new Go release comes along.

  [1]: https://golang.org/doc/devel/release.html#policy

Users should be able to add `go: [oldstable, stable]` to their .travis.yml
once this is released.

abhinav committed:
Support installing the previous stable release

This adds support for installing the previous stable release of Go with
`gimme previous`.

Coupled with `gimme stable`, this will make it possible to easily
support the current and previous stable releases of Go. This matches
[Go's Release Policy][1]: the current and previous stable releases are
supported until a new Go release comes along.

  [1]: https://golang.org/doc/devel/release.html#policy

akshayjshah commented on gimme:
```diff
@@ -655,6 +669,16 @@ _update_stable() {
+	local prev_x
+	prev_x=$(_get_curr_stable | awk -F. '{
+		$2--;
+		print $1 "." $2 "." "x"
```
Nice! Reusing the resolver for the `1.10.x` notation is clever.

akshayjshah commented:
Friends at Travis, is anyone available to take a look at this PR? If you'd prefer that we solve this problem differently (or not at all), please let us know!

theckman commented:
`previous` is a bit ambiguous to me, even as a native English speaker. If this were ran on `1.11.4` (when it was `stable`) would `previous` pull `1.11.3` or `1.10.8`? Borrowing from Debian, would `oldstable` make more sense? I only ask because depending on which functionality this supports, I could see adding functionality for the other workflow.

abhinav commented:
`oldstable` makes sense. I'll switch it over.

abhinav committed:
gimme/previous: Renable to oldstable

This  renames the `previous` alias to `oldstable` which is more obvious
than `previous` as to what the behavior will be.

abhinav committed:
Merge branch 'master' into previous-stable

akshayjshah commented:
(Edit, realized that @theckman is another interested party but not a maintainer.)

Friends from Travis, can someone please take a look at this PR?

BanzaiMan commented:
Thanks for the PR. This looks good. Please update the README as well.

abhinav committed:
Merge branch 'master' into previous-stable

abhinav committed:
README: Document oldstable

This adds information about `oldstable` in the README in all places
where `stable` is documented.

abhinav commented:
> Thanks for the PR. This looks good. Please update the README as well.

README updated. Thanks!

akshayjshah commented:
Thanks @BanzaiMan! When should we expect this to be usable in Travis builds?
